content/src/main/content/jcr_root/apps/acs-commons/authoring/shared-component-properties-classic/shared-component-properties-classic.js

```diff
@@ -0,0 +1,105 @@
+                                    editActionIdx = i;
+                                    break;
+                                }
+                            }
+                            if (editActionIdx !== null) {
+                                config.actions.splice(editActionIdx + 1, 0, CQ.wcm.EditBase.EDITGLOBAL);
+                                config.actions.splice(editActionIdx + 1, 0, CQ.wcm.EditBase.EDITSHARED);
+                            }
+                        }
+                    }
+                }
+                CQ.wcm.EditBase.applyConfigDefaults.call(this, config, defaults);
+            }
+        });
+        CQ.Ext.reg(xtype, newType);
+    };
+
+    // Define the handlers for CQ.wcm.EditBase.EDITGLOBAL and CQ.wcm.EditBase.EDITSHARED
+    var editSharedHandler = function(type, editBaseType) {
+        var origDialog = this.dialog;
+        var origDialogCached = this.dialogs[CQ.wcm.EditBase.EDIT];
+        var origPath = this.path;
+
+        var pageInfo = CQ.utils.WCM.getPageInfo(this.path);
+        var newPath = pageInfo.sharedComponentProperties.root + "/jcr:content/" + type + "-component-properties";
+        if (type == "shared") {
+            newPath += "/" + this.getResourceType();
+        }
+
+        try {
+            this.dialog = this.dialog + "_" + type;
+            this.dialogs[CQ.wcm.EditBase.EDIT] = this.dialogs[editBaseType];
+            this.path = newPath;
+
+            // For global props, need to unregister a previously fetched global
+            // props dialog since it always has the same data path but the
+            // dialog itself can be different for different components.
+            if (type == "global") {
+                CQ.WCM.unregisterDialog("editdialog-" + this.path);
+            }
+
+            CQ.wcm.EditBase.showDialog(this, editBaseType);
+        } catch (e) {
+            if (window.console && console.error) {
+                console.error(e);
+            }
+        } finally {
+            this.dialog = origDialog;
+
+            // Don't allow the global props dialog to be cached, as it causes
+            // problems due to always having the same data path.
+            if (type != "global") {
+                this.dialogs[editBaseType] = this.dialogs[CQ.wcm.EditBase.EDIT];
+            }
+
+            this.dialogs[CQ.wcm.EditBase.EDIT] = origDialogCached;
+            this.path = origPath;
+        }
+    };
+
+    function addSharedConfigActionHandlers(clazz) {
+        clazz.ActionsConvertor[CQ.wcm.EditBase.EDITSHARED] = {
+            text: CQ.I18n.getMessage("Edit Shared"),
+            handler: function (evt) {
+                // Using 'evt.ownerCt.defaults.scope' instead of 'this' because
+                // 'this' has a permanent reference to the first component for
+                // which the menu is launched.
+                editSharedHandler.call(evt.ownerCt.defaults.scope, "shared", CQ.wcm.EditBase.EDITSHARED);
```

Comment:
Really crappy scoping issue here where I couldnt use `this` because of how ActionsConvertors are singletons.